bots/DiscordBot/PublicCommands.js

```diff
@@ -0,0 +1,357 @@
+                    reaction.emoji.name === '⬅️' ? currentPage -= 1 : currentPage += 1;
+                    if (currentPage < 0) currentPage = 4;
+                    if (currentPage > 4) currentPage = 0;
+                    message.edit(generateResponse(focusArray[currentPage]));
+                    try {
+                        await message.react('⬅️')
+                        await message.react('➡️')
+                    } catch (error) {
+                        console.log('One of the emojis failed to react:', error);
+                    }
+                });
+            });
+        });
+    }
+
+    async help(msg, directions) {
+        //!help <command>
+        let id = msg.author.id;
+        let response = `Hey <@${id}>,\n`;
+        if (directions.length == 0) {
+            response += this.botResponseService.helpMain;
+            msg.channel.send(response);
+        } else {
+            switch (directions[0]) {
+                case 'gameinfo':
+                    response += this.botResponseService.helpGameinfo;
+                    msg.channel.send(response);
+                    break;
+                case 'help':
+                    response += this.botResponseService.helpHelp;
+                    msg.channel.send(response);
+                    break;
+                case 'leaderboard_global':
+                    response += this.botResponseService.helpLeaderboard_global
+                    msg.channel.send(response);
+                    break;
+                case 'leaderboard_local':
+                    response += this.botResponseService.helpLeaderboard_local
+                    msg.channel.send(response);
+                    break;
+                case 'userinfo':
+                    response += this.botResponseService.helpUserinfo
+                    msg.channel.send(response);
+                    break;
+                default:
+                    response += this.botResponseService.helpUnidentified
+                    msg.channel.send(response);
+            }
+        }
+    }
+
+    async leaderboard_global(msg, directions) {
+        //!leaderboard_global <filter> (<page>)
+
+        // Calculating how the leaderboard looks
+        let sortingKey = directions[0];
+        let limit = await this.userService.getUserCount();
+        let result = await this.leaderboardService.getLeaderboard(limit, sortingKey);
+        let leaderboard = result.leaderboard;
+        let pageCount = Math.ceil(leaderboard.length / 20)
+
+        // here be dragons
+        const getNestedObject = (nestedObj, pathArr) => {
+            return pathArr.reduce((obj, key) =>
+                (obj && obj[key] !== 'undefined') ? obj[key] : -1, nestedObj)
+        }
+
+        const generateLeaderboard = page => {
+            let position_list = "";
+            let username_list = "";
+            let sortingKey_list = "";
+            let lowerLimit = (page - 1) * 20 + 1
+            let upperLimit = page * 20
```

Comment:
Should be able to work out how many pages can be supported by running a `count` on the leaderboard?